sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/_artifacts/_package/model_package.py

```diff
@@ -0,0 +1,292 @@
+from azure.ai.ml.entities._util import load_from_dict
+from azure.ai.ml.constants._common import (
+    BASE_PATH_CONTEXT_KEY,
+    PARAMS_OVERRIDE_KEY,
+)
+from azure.ai.ml._utils.utils import snake_to_pascal, dump_yaml_to_file
+from azure.ai.ml._utils._experimental import experimental
+
+
+@experimental
+class PackageInputPathId:
+    def __init__(self, input_path_type: Optional[str] = None, resource_id: Optional[str] = None):
+        """PackageInputPathId.
+
+        :param input_path_type: The type of the input path.
+        :type input_path_type: str
+        :param resource_id: The resource id of the input path.
+        :type resource_id: str
+        """
+        self.input_path_type = input_path_type
+        self.resource_id = resource_id
+
+    def _to_rest_object(self) -> RestPackageInputPathId:
+        return RestPackageInputPathId(
+            input_path_type=self.input_path_type,
+            resource_id=self.resource_id,
+        )
+
+    @classmethod
+    def _from_rest_object(cls, package_input_path_id_rest_object: RestPackageInputPathId) -> "PackageInputPathId":
+        return PackageInputPathId(
+            input_path_type=package_input_path_id_rest_object.input_path_type,
+            resource_id=package_input_path_id_rest_object.resource_id,
+        )
+
+
+@experimental
+class PackageInputPathVersion:
+    def __init__(
+        self,
+        input_path_type: Optional[str] = None,
+        resource_name: Optional[str] = None,
+        resource_version: Optional[str] = None,
+    ):
+        """PackageInputPathVersion.
+
+        :param input_path_type: The type of the input path.
+        :type input_path_type: str
+        :param resource_name: The resource name of the input path.
+        :type resource_name: str
+        :param resource_version: The resource version of the input path.
+        :type resource_version: str
+        """
+        self.input_path_type = input_path_type
+        self.resource_name = resource_name
+        self.resource_version = resource_version
+
+    def _to_rest_object(self) -> RestPackageInputPathVersion:
+        return RestPackageInputPathVersion(
+            input_path_type=self.input_path_type,
+            resource_name=self.resource_name,
+            resource_version=self.resource_version,
+        )
+
+    @classmethod
+    def _from_rest_object(
+        cls, package_input_path_version_rest_object: RestPackageInputPathVersion
+    ) -> "PackageInputPathVersion":
+        return PackageInputPathVersion(
+            input_path_type=package_input_path_version_rest_object.input_path_type,
+            resource_name=package_input_path_version_rest_object.resource_name,
+            resource_version=package_input_path_version_rest_object.resource_version,
+        )
+
+
+@experimental
+class PackageInputPathUrl:
+    def __init__(self, input_path_type: Optional[str] = None, url: Optional[str] = None):
+        """PackageInputPathUrl.
```

Comment:
"Package input path URL"
I see in other places we don't put the class name in the docstring but an English description of what the class is